javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -79,1587 +91,989 @@ class Formatter(
-      var result                = new StringBuilder()
-      val bigInt                = arg.asInstanceOf[BigInteger]
-      val currentConversionType = formatToken.getConversionType()
-
-      if (formatToken.isFlagSet(FormatToken.FLAG_MINUS) || formatToken
-            .isFlagSet(FormatToken.FLAG_ZERO)) {
-        if (!formatToken.isWidthSet())
-          throw new MissingFormatWidthException(formatToken.getStrFlags())
-      }
-
-      if (formatToken.isFlagSet(FormatToken.FLAG_ADD) && formatToken.isFlagSet(
-            FormatToken.FLAG_SPACE))
-        throw new IllegalFormatFlagsException(formatToken.getStrFlags())
-
-      if (formatToken.isFlagSet(FormatToken.FLAG_ZERO) && formatToken
-            .isFlagSet(FormatToken.FLAG_MINUS))
-        throw new IllegalFormatFlagsException(formatToken.getStrFlags())
-
-      if (formatToken.isPrecisionSet())
-        throw new IllegalFormatPrecisionException(formatToken.getPrecision())
 
-      if ('d' != currentConversionType && formatToken.isFlagSet(
-            FormatToken.FLAG_COMMA))
-        throw new FormatFlagsConversionMismatchException(
-          formatToken.getStrFlags(),
-          currentConversionType)
+      case 'c' | 'C' =>
+        validateFlags(flags,
+                      conversion,
+                      invalidFlags = NumericOnlyFlags | AltFormat)
+        rejectPrecision()
+        def checkValidCodePoint(arg: Int): Unit =
+          if (!Character.isValidCodePoint(arg))
+            throw new IllegalFormatCodePointException(arg)
 
-      if (formatToken.isFlagSet(FormatToken.FLAG_SHARP) && 'd' == currentConversionType)
-        throw new FormatFlagsConversionMismatchException(
-          formatToken.getStrFlags(),
-          currentConversionType)
+        def formatCharString(charString: String) =
+          formatNonNumericString(localeInfo, flags, width, -1, charString)
 
-      if (null == bigInt)
-        return transformFromNull()
-
-      isNegative = (bigInt.compareTo(BigInteger.ZERO) < 0)
-
-      if ('d' == currentConversionType) {
-        val numberFormat = getNumberFormat()
-        val readableName = formatToken.isFlagSet(FormatToken.FLAG_COMMA)
-        numberFormat.setGroupingUsed(readableName)
-        result.append(numberFormat.format(bigInt))
-      } else if ('o' == currentConversionType) {
-        result.append(bigInt.toString(8))
-      } else {
-        result.append(bigInt.toString(16))
-      }
-      if (formatToken.isFlagSet(FormatToken.FLAG_SHARP)) {
-        startIndex = if (isNegative) 1 else 0
-        if ('o' == currentConversionType) {
-          result.insert(startIndex, "0")
-          startIndex += 1
-        } else if ('x' == currentConversionType || 'X' == currentConversionType) {
-          result.insert(startIndex, "0x")
-          startIndex += 2
+        arg match {
+          case arg: Byte =>
+            checkValidCodePoint(arg)
+            formatCharString(arg.toChar.toString)
+
+          case arg: Char =>
```

Comment:
Either remove those blank lines here, or add one before `case arg: Int`. Otherwise it's not consistent, and it looks like `Short` and `Int` are more related than the other ones.